File: CPP [3501-4000]/3627.maximum_median_sum_of_subsequences_of_size_3 [1].cpp
class Solution {
public:
    // T.C.=O(n*log(n)), S.C.=O(1)
    // Greedy
    long long maximumMedianSum(vector<int>& nums) {
        sort(nums.begin(), nums.end());

        long long int sum = 0;
        for(int i=0, j=nums.size()-1; i<j; i++, j-=2)
            sum += nums[j-1];

        return sum;
    }
};

/*
it is optimal to take the 2 largest elements (say, B and C) 
and 1 smallest element (say, A) thus forming the sequence A,B,C
the median B is added to the answer until no more size 3 sequence is left

# similar: [1561. maximum-number-of-coins-you-can-get]
*/

File: Javascript/3627.maximum_median_sum_of_subsequences_of_size_3 [1].js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maximumMedianSum = function(nums) {
    nums.sort((a, b) => a-b)

    let sum = 0
    for(let i=0, j=nums.length-1; i<j; i++, j-=2)
        sum += nums[j-1]

    return sum
};